src/targets/__tests__/awsLambda.test.ts

```diff
@@ -0,0 +1,216 @@
+    getAwsLambdaTarget();
+  });
+});
+
+describe('publish', () => {
+  beforeAll(() => {
+    setAwsEnvironmentVariables();
+  });
+
+  const noArtifactsForRevision = jest.fn().mockImplementation(function() {
+    return [];
+  });
+
+  test('error on missing zip file', async () => {
+    const awsTarget = getAwsLambdaTarget();
+    awsTarget.getArtifactsForRevision = noArtifactsForRevision.bind(
+      AwsLambdaTarget
+    );
+    // `publish` should report an error. When it's not dry run, the error is
+    // thrown; when it's on dry run, the error is logged and `undefined` is
+    // returned. Thus, both alternatives have been considered.
+    try {
+      const noPackageFound = await awsTarget.publish('version', 'revision');
+      expect(noPackageFound).toBe(undefined);
+    } catch (error) {
+      expect(error instanceof Error).toBe(true);
+    }
+  });
+
+  const twoArtifactsForRevision = jest.fn().mockImplementation(function() {
+    return ['file1', 'file2'];
+  });
+
+  test('error on having too many files', async () => {
+    const awsTarget = getAwsLambdaTarget();
+    awsTarget.getArtifactsForRevision = twoArtifactsForRevision.bind(
+      AwsLambdaTarget
+    );
+    // `publish` should report an error. When it's not dry run, the error is
+    // thrown; when it's on dry run, the error is logged and `undefined` is
+    // returned. Thus, both alternatives have been considered.
+    try {
+      const tooManyPackagesFound = await awsTarget.publish(
+        'version',
+        'revision'
+      );
+      expect(tooManyPackagesFound).toBe(undefined);
+    } catch (error) {
+      expect(error instanceof Error).toBe(true);
+    }
+  });
+
+  const singleArtifactsForRevision = jest
+    .fn()
+    .mockImplementation(function(input: string) {
+      return [input];
+    });
+
+  const downloadArtifactMock = jest
+    .fn()
+    .mockImplementation(function(input: string) {
+      return input;
+    });
+
+  const publishAwsLayerMock = jest.fn().mockImplementation(function() {
+    return {
+      Version: 1,
+      LayerVersionArn: 'layer:version:arn:test',
+    };
+  });
+  const publishAwsLayerMockUndefinedVersion = jest
+    .fn()
+    .mockImplementation(function() {
+      return {
+        Version: undefined,
+        LayerVersionArn: 'layer:version:arn:test',
+      };
+    });
+
+  const addLayerPermissionsMock = jest.fn().mockImplementation(function() {
+    // Do nothing
+  });
+
+  test('success on publishing', async () => {
+    const awsTarget = getAwsLambdaTarget();
+    awsTarget.getArtifactsForRevision = singleArtifactsForRevision.bind(
+      AwsLambdaTarget
+    );
+    awsTarget.artifactProvider.downloadArtifact = downloadArtifactMock.bind(
+      awsTarget
+    );
+    awsTarget.publishAwsLayer = publishAwsLayerMock.bind(AwsLambdaTarget);
+    awsTarget.addAwsLayerPermissions = addLayerPermissionsMock.bind(
+      AwsLambdaTarget
+    );
+    awsTarget.publish('', '');
```

Comment:
After running `publish` we should check if it actually called those mocked functions with right arguments. Also, `publish` is async, so you should await on it at some point.